src/main/java/ch/njol/skript/effects/EffTeamOption.java

```diff
@@ -0,0 +1,114 @@
+package ch.njol.skript.effects;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.bukkit.scoreboard.Team;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.UnknownNullability;
+
+// todo doc
+public class EffTeamOption extends Effect {
+
+	public static final Team.OptionStatus[] OPTIONS = Team.OptionStatus.values();
+
+	public static String optionStatus() {
+		// test-only; make sure the enum erasure hasn't changed between versions
+		assert OPTIONS[0] == Team.OptionStatus.ALWAYS
+			&& OPTIONS[1] == Team.OptionStatus.NEVER
+			&& OPTIONS[2] == Team.OptionStatus.FOR_OTHER_TEAMS
+			&& OPTIONS[3] == Team.OptionStatus.FOR_OWN_TEAM;
+		return "(3:this team|2:other teams|0:every(body|one)|1:no(body|one))";
+	}
+
+	public static String statusName(Team.OptionStatus status) {
+		switch (status) {
+			case ALWAYS:
+				return "everybody";
+			case NEVER:
+				return "nobody";
+			case FOR_OTHER_TEAMS:
+				return "other teams";
+			default:
+				return "this team";
+		}
+	}
+
+	static {
+		String optionStatus = optionStatus();
+		Skript.registerEffect(EffTeamOption.class,
+			"(allow|permit) " + optionStatus + " to (collide:collide with|tag:see [the] name[ ]tag[s] of|death:see the death message[s] of) %teams%",
+				"(allow|permit) " + optionStatus + " to see %teams%'[s] (tag:name[ ]tag[s]|death:death message[s])",
+				"(allow|permit) friendly fire for %teams%",
+				"(prevent|deny) friendly fire for %teams%",
+				"(allow|permit) %teams% to see [friendly] invisible[ player]s",
+				"(prevent|deny) %teams% from seeing [friendly] invisible[ player]s"
```

Comment:
```suggestion
				"(allow|permit) %teams% to see friendly invisible[ player]s",
				"(prevent|deny) %teams% from seeing friendly invisible[ player]s"
```
I think making friendly optional implies that you can see all invisible players, when that's not possible

Reply:
Yeah I worried about that too, I was just concerned it would be complicated and long otherwise